Add new msvs_toolset (#2951)
conan-io/docs: reference/conanfile/tools/microsoft.rst

        check_min_vs(self, "192")

.. _conan_tools_microsoft_msvs_toolset:

msvs_toolset()
--------------

Available since: `1.59.0 <https://github.com/conan-io/conan/releases/tag/1.59.0>`_

.. code-block:: python

    def msvs_toolset(conanfile)

Returns the corresponding Visual Studio or msvs platform toolset based on the settings of the given ``conanfile``. For instance, it may return ``v143``

> for ``compiler=Visual Studio`` with ``compiler.version=17``. If ``compiler.toolset`` was set in settings, it has a priority and always returned.

Reviewer: what does it return for other compilers?

Author: An empty string for any other cases. I just added it to the description now. Good catch.

Parameters:
    - ``conanfile``: Always use ``self``, the current recipe

Example:

.. code-block:: python

    from conan.tools.microsoft import msvs_toolset

    def validate(self):

>         msvs_toolset(self)

Reviewer: this example doesn't use a returned value?

Author: updated to use the returned value

        check_min_vs(self, "192")

.. _conan_tools_microsoft_msvs_toolset:

msvs_toolset()
--------------

Available since: `1.59.0 <https://github.com/conan-io/conan/releases/tag/1.59.0>`_

.. code-block:: python

    def msvs_toolset(conanfile)

Returns the corresponding Visual Studio or msvs platform toolset based on the settings of the given ``conanfile``. For instance, it may return ``v143``
for ``compiler=Visual Studio`` with ``compiler.version=17``. If ``compiler.toolset`` was set in settings, it has a priority and always returned.

> In case of invalid compiler or invalid version, it return an empty string.

Author: ```suggestion
In case of invalid compiler or invalid version, ``msvs_toolset`` returns an empty string.
```

Parameters:
    - ``conanfile``: Always use ``self``, the current recipe

Example:

.. code-block:: python

>     from conan.tools.microsoft import msvs_toolset

Author: ```suggestion
    from conan.tools.microsoft import msvs_toolset
    from conan.tools.scm import Version
    from conan.errors import ConanInvalidConfiguration
```

    def validate(self):
        toolset = msvs_toolset(self)

>         if Version(toolset) < "v110":

Reviewer: ```suggestion
        if Version(toolset) < "v110":
```
it needs an import

Author: done.
